Update reliability documentation

lukasfritsch opened the pull request:
Update the documentation of the `Reliability` manual. Adding some descriptions and examples along with references.

lukasfritsch commented on docs/src/manual/reliability.md:
````diff
@@ -1,20 +1,261 @@
+println("Coefficient of variation: $(std_is/pf_is)")
+```
+
+Since Importance Sampling relies on FORM, the results are similar to those of FORM.
````
This is probably not a good example to also show how FORM and Importance Sampling (should) work. I originally used it for Line Sampling and there it works pretty well. 

Thus it's probably the best if I choose a different example that works for all methods

AnderGray replied:
I might be okey to keep it like this, if this is just what is it, and IS is a poor estimator in this case. Do we know when IS is a good thing to do? If so, we can mention this here, and leave it as is. E.g., FORM gives exact answer when Limit state is linear. When does IS give good answers?

lukasfritsch commented on docs/src/manual/reliability.md:
```diff
@@ -1,20 +1,261 @@
+This approach involves decomposing the problem into a sequence of conditional probabilities that are estimated using Markov Chain Monte Carlo.
 
-with the failure domain
+Here we solve a simple problem taken from [zuevSubsetSimulationMethod2013](@cite) where the response $y(\boldsymbol{X})$ depends on two independent random variables $X_1$ and $X_2$ following a standard normal distribution. The simple linear model is defined by
```
I kept the original example used for Subset there.
But you're right, it makes more sense to stick to one example

lukasfritsch commented:
Thanks a lot for the feedback @AnderGray , I implemented the requested changes.

I changed the example to one that works better with all methods and also added that example in the `demo` folder. That one is now used for all methods, also subset.
I also noted a typo in one in a print statement in  `demo\reliability\linesampling.jl`and changed that.

AnderGray reviewed (approved):
Great stuff. The new demo is nice, and good spot with the line sample demo error. Added a small missing space in the note.

I think good to go after a merge with master.
